cmd/notary/keys.go

```diff
@@ -580,30 +579,43 @@ func (k *keyCommander) keyPassphraseChange(cmd *cobra.Command, args []string) er
 		return fmt.Errorf("invalid key ID provided: %s", keyID)
 	}
 
-	// Find the key's GUN by ID, in case it is a non-root key
-	var keyGUN string
-	for _, store := range ks {
-		for keypath := range store.ListKeys() {
-			if filepath.Base(keypath) == keyID {
-				keyGUN = filepath.Dir(keypath)
-			}
+	// Find which keyStore we should replace the key password in, and replace if we find it
+	var foundKeyStore trustmanager.KeyStore
+	var privKey data.PrivateKey
+	var keyInfo trustmanager.KeyInfo
+	var cs *cryptoservice.CryptoService
+	for _, keyStore := range ks {
+		cs = cryptoservice.NewCryptoService(keyStore)
+		if privKey, _, err = cs.GetPrivateKey(keyID); err == nil {
+			foundKeyStore = keyStore
+			break
 		}
 	}
-	cs := cryptoservice.NewCryptoService(keyGUN, ks...)
-	privKey, role, err := cs.GetPrivateKey(keyID)
-	if err != nil {
+	if foundKeyStore == nil {
 		return fmt.Errorf("could not retrieve local key for key ID provided: %s", keyID)
 	}
-
 	// Must use a different passphrase retriever to avoid caching the
 	// unlocking passphrase and reusing that.
 	passChangeRetriever := k.getRetriever()
-	keyStore, err := trustmanager.NewKeyFileStore(config.GetString("trust_dir"), passChangeRetriever)
-	err = keyStore.AddKey(filepath.Join(keyGUN, keyID), role, privKey)
+	var addingKeyStore trustmanager.KeyStore
+	switch foundKeyStore.Name() {
+	case "yubikey":
```

Comment:
Totally non-blocking, and just because you previously mentioned you didn't like switching on the name very much - what do you think about having the keystore return constant types sort of like key algorithms are constants?  Then here we can switch based on the constants.  

(this may not be worth doing at all, since we may want to refactor all this at a later point anyway)